core/src/Kokkos_CopyViews.hpp

```diff
@@ -2918,7 +2918,15 @@ inline void deep_copy(
   }
 #endif
 
-  if (dst.data() == nullptr || src.data() == nullptr) {
+  dst_value_type* dst_start = dst.data();
+  dst_value_type* dst_end   = dst.data() + dst.span();
+  src_value_type* src_start = src.data();
+  src_value_type* src_end   = src.data() + src.span();
+
+  // Early dropout if identical range
+  if ((dst_start == nullptr || src_start == nullptr) ||
+      ((std::ptrdiff_t(dst_start) == std::ptrdiff_t(src_start)) &&
+       (std::ptrdiff_t(dst_end) == std::ptrdiff_t(src_end)))) {
```

Comment:
This is now in inbalance with the overload that doesn't take an execution space https://github.com/kokkos/kokkos/blob/8bbcd0ba84842810d15f78b17901da60471f911c/core/src/Kokkos_CopyViews.hpp#L1751-L1755. Can we modify it accordingly?

Reply:
Lets do this as part of a proper code duplication reduction exercise in the next release when we also can delete deprecated stuff. Originally I only wanted to address too much fencing in the explicitly supposed asynchronous deep copy call. I am fixing the issue Damien found here in the correct way since we do not need to repeat the slightly off way it is done in the other deep_copy call (it would allow a deep copy no op of `deep_copy(View<double[2][6]>(ptr),View<double[6][2]>(ptr))` since the ranges are the same - even though semantically these are two different views. But again lets just get rid of that by later correctly calling the three argument version from the two argument version.